src/main/scala/org/kframework/minikore/MiniKoreInterface.scala

```diff
@@ -258,145 +197,119 @@ object PatternInterface {
 
     override val _2 = q
 
+    override def apply(p: Pattern, q: Pattern): Or
   }
 
-  trait OrBuilder extends Node2Builder[Pattern] {
-
-    override def apply(x: Pattern, y: Pattern): Or
-
-  }
 
   object Or {
-    def unapply(arg: Pattern): Option[(Pattern, Pattern)] = arg match {
-      case v: Or => Some(v.p, v.q)
-      case _ => None
-    }
+    def unapply(arg: Or): Option[(Pattern, Pattern)] = Some(arg.p, arg.q)
   }
 
-  trait Not extends Pattern with Node1[Pattern] {
+
+  trait Not extends Pattern with Node1 {
     def p: Pattern
 
     override val _1 = p
 
-  }
-
-  trait NotBuilder extends Node1Builder[Pattern] {
     override def apply(p: Pattern): Not
-
   }
 
+
   object Not {
-    def unapply(arg: Pattern): Option[Pattern] = arg match {
-      case n: Not => Some(n.p)
-      case _ => None
-    }
+    def unapply(arg: Not): Option[Pattern] = Some(arg.p)
   }
 
-  trait Application extends Pattern with LabelledNode[String, Pattern] {
 
-    def args: Seq[_ <: Pattern]
+  trait Application extends Pattern with LabeledNode[String] {
+    def args: Seq[Pattern]
 
     override def children: Seq[Pattern] = args
 
-
+    override def apply(label: String, children: Seq[Pattern]): Application
   }
 
-  trait ApplicationBuilder extends LabelledNodeBuilder[String, Pattern] {
-    override def apply(x: String, c: Seq[_ <: Pattern]): Application
-  }
 
   object Application {
-    def unapply(arg: Pattern): Option[(String, Seq[_ <: Pattern])] = arg match {
+    def unapply(arg: Pattern): Option[(String, Seq[Pattern])] = arg match {
       case a: Application => Some(a.label, a.args)
       case _ => None
     }
   }
 
-  trait Implies extends Pattern with Node2[Pattern] {
+
+  trait Implies extends Pattern with Node2 {
     def p: Pattern
 
     def q: Pattern
 
     override val _1 = p
 
     override val _2 = q
-  }
 
-  trait ImpliesBuilder extends Node2Builder[Pattern] {
     override def apply(p: Pattern, q: Pattern): Implies
   }
 
+
   object Implies {
-    def unapply(arg: Pattern): Option[(Pattern, Pattern)] = arg match {
-      case v: Implies => Some(v.p, v.q)
-      case _ => None
-    }
+    def unapply(arg: Implies): Option[(Pattern, Pattern)] = Some(arg.p, arg.q)
   }
 
-  trait Exists extends Pattern with Node2[Pattern] {
+
+  trait Exists extends Pattern with Node2 {
     def v: Variable
 
     def p: Pattern
 
     override val _1 = v
 
     override val _2 = p
+
+    def apply(v: Variable, p: Pattern): Exists
+
+    override def apply(p: Pattern, q: Pattern): Exists = apply(v, q)
```

Comment:
This `apply` does not look right: takes `p` and `q` and uses only `q` (and `v`?).  Fix and add a test that would catch this error.